swap/swap_test.go

```diff
@@ -1818,6 +1821,83 @@ func TestPeerGetLastSentCumulativePayout(t *testing.T) {
 	}
 }
 
+func TestAvailableBalance(t *testing.T) {
+	testBackend := newTestBackend()
+	defer testBackend.Close()
+	swap, clean := newTestSwap(t, ownerKey, testBackend)
+	defer clean()
+	cleanup := setupContractTest()
+	defer cleanup()
+
+	// deploy a chequebook
+	err := testDeploy(context.TODO(), swap, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// create a peer
+	peer, err := swap.addPeer(newDummyPeerWithSpec(Spec).Peer, swap.owner.address, swap.GetParams().ContractAddress)
+
+	// deposit 100
+	depositAmount := big.NewInt(100)
+	opts := bind.NewKeyedTransactor(swap.owner.privateKey)
+	opts.Value = depositAmount
+	opts.Context = context.TODO()
+	rec, err := swap.contract.Deposit(opts, swap.backend)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if rec.Status != types.ReceiptStatusSuccessful {
```

Comment:
ok, it turns out i was wrong in my last comment regarding this part of the test.

it seems `waitForTx` already checks this:
```golang
// indicate whether the transaction did not revert
if receipt.Status != types.ReceiptStatusSuccessful {
	return nil, ErrTransactionReverted
}
```
so there's no need to check the receipt outside of this call.

my bad, apologies

Reply:
I would like to keep it. I spend a significant time debugging yesterday, as `testWaitForTx` did not include the snippet which you posted above (I added it already). Having this additional check would have prevented me from spending so much time debugging.

Reply:
ok, so it's awesome that you added this to `testWaitForTx`

but since it's there already, why also have it in `TestAvailableBalance`? what value is it adding?

Reply:
A person could change testWaitForTX, for example, and change something in the code as well. 

It is not super important for me, but I just think we can as well leave it in.